src/lucky_record/associations.cr

```diff
@@ -9,12 +9,21 @@ module LuckyRecord::Associations
 
   macro belongs_to(type_declaration)
     {% assoc_name = type_declaration.var }
-    {% model = type_declaration.type %}
 
-    field {{ assoc_name.id }}_id : Int32
+    {% if type_declaration.type.is_a?(Union) %}
+      {% model = type_declaration.type.types.first %}
+      {% nilable = true %}
+    {% else %}
+      {% model = type_declaration.type %}
+      {% nilable = false %}
+    {% end %}
+
+    field {{ assoc_name.id }}_id : Int32{% if nilable %}?{% end %}
 
     def {{ assoc_name.id }}
-      {{ model }}::BaseQuery.new.find({{ assoc_name.id }}_id)
+      {{ assoc_name.id }}_id.try do |value|
+        {{ model }}::BaseQuery.new.find({{ assoc_name.id }}_id)
```

Comment:
Oh and I thin this should use the `value` in the `find`. Just to make sure it uses the not nil version: `find(value)`. That was my bad. I think I did it wrong in the snippet I posted earlier

Reply:
Ah yes, you mean like this?
```crystal
{{ assoc_name.id }}_id.try do |value|
  {{ model }}::BaseQuery.new.find(value)
end
```

Reply:
Yep!